### PR TITLE
Make fetchValue, getValue and keyAlreadyExists faster and more swifty

### PR DESCRIPTION
**Pull request by janakmshah**
TLDR:
Take advantage of Swift's `Array.contains` and `Dictionary subscript` which return optionals. 
Use `guard` instead of large `if` blocks.
Replace inefficient `for loops` with Swift's array features.

------

The two functions `getValue` and `fetchValue` currently takes 3 steps:

1. Convert plist into Dictionary
2. Loop through Dictionary keys to find the inputted key
3. Get the value for the given key

At the moment step 2 is inefficient because it continues looping through the entire key array, even after we're found the key we're looking for.
So we could break out of this loop when we find the given key.
However, we can take it one step further and remove this step all together because `dict[key]` will return us the value if the key exists or it will return nil if the key doesn't exist. Combining steps 2 & 3 into one, more efficient, step.

------

The function `keyAlreadyExists`, again continues looping and taking up resources even after we've identified that the key does exist.
Rather than using a `for loop` to match the key String, we can use Swift's `Array.contains` function which will tell us if the key exists, or return nil if it does not :)

### CODE REVIEW

**Comment by rebeloper** [comment]
Thanks for the update
